### PR TITLE
Deploy pipeline by using deployment manager

### PR DESCRIPTION
**Pull request by akm**
- Implement Builder Process method
  - Call deployment manager to build pipeline
    - The pipeline doesn't include managed instance group yet.
- Add ".json" extension to request path
- Add the following paths for HTML pages
  - GET /pipelines.html
  - POST /pipelines.html
  - GET /pipelines/new.html

## Reviewers

- @nagachika 
- @taigou 
- @ko1nksm 
- @guemon 
- @totty 
- @minimum2scp

### CODE REVIEW

File: builder.go [hunk 1]
```diff
@@ -1,14 +1,124 @@
...
+				Type: "pubsub.v1.topic",
+				Name: topic,
+				Properties: map[string]interface{}{ "topic": topic },
+			},
+			Resource{
+				Type: "pubsub.v1.subscription",
+				Name: subscription,
+				Properties: map[string]interface{}{
+					"subscription": subscription,
+					"topic": fmt.Sprintf("projects/%s/topics/%s", project, topic),
```
Comment (nagachika) [inline builder.go]: Just pass the reference to the topic resource is also create dependency between subscription and topic.
See https://cloud.google.com/deployment-manager/docs/configuration/syntax-reference#references
In this context, you can use `fmt.Sprintf("$(ref.%s.name)")` to reference the topic created above and ommit `DependsOn` metadata.

**Comment by nagachika** [comment]
I've approved this PR. This is just for your information.

JSON is compatible to YAML, I mean the data encoded in JSON format can be decoded as YAML perfectly.
So you can encode the resources to be passed to Deployment Manager API in JSON format.
This allow you to depend on standard package `encoding/json` instead of third party packages.
If there're any difficulties with these third party packages, remember this comment.

File: builder.go (hunk 1 above)
Comment (akm) [inline builder.go]: @nagachika I fixed this. I removed Metadata struct also. Thank you!

**Comment by akm** [comment]
@nagachika I wasn't good at imagining YAML loader can read JSON documents, but now I can. I removed the dependency to 3rd party package. Thanks a lot!

**Pull request merged**